var/spack/repos/builtin/packages/r/package.py

```diff
@@ -168,6 +171,9 @@ def configure_args(self):
         if '+memory_profiling' in spec:
             config_args.append('--enable-memory-profiling')
 
+        if '+libpng' in spec:
+            config_args.append('--with-libpng')
```

Comment:
@jasonbuechler This case is so frequent in packaging (having a variant that triggers a `--with/--without` at configure time) that we have a one line convenience function for that:
```suggestion
        config_args += self.with_or_without('libpng')
```